### DISCUSSION TITLE
Using the constant function

### DISCUSSION THREAD

**Discussion by vinicius98s** [gcanti/fp-ts#1735] [Q&A] [answered]
Hello, I recently found the [constant](https://github.com/gcanti/fp-ts/blob/master/src/function.ts#L157-L159) function, and I noticed a weird behavior (at least from my point of view, so I might have misunderstood its usage).
So, if you run the following snippet:
```ts
import * as O from "fp-ts/Option";
import { constant, pipe } from "fp-ts/function";

pipe(
  O.some("hello"),
  O.fold(constant(console.log("none")), constant(console.log("some")))
)
```
You will notice that it outputs both "none" and "some" in the terminal.
I thought it would be the same as:
```ts
import * as O from "fp-ts/Option";
import { pipe } from "fp-ts/function";

pipe(
  O.some("hello"),
  O.fold(() => console.log("none"), () => console.log("some"))
)
```
Which just outputs "some", I'm still a little bit confused on why this happens because the `constant` function returns `Lazy<A>` function, can someone clarify this?
Thank you

**Comment by david-zacharias** [comment, marked as answer]
I'd say, you nearly gave the answer by your self: `constant` function **returns** `Lazy<A>`. In your first example the `console.log("none")` and  `console.log("some")` will be executed instantly and their return values (`void` in this case) becomes lazy.

**Reply by samhh** [reply]
It should be noted that `console.log` can be purely represented with `IO`, as per the [`Console`](https://gcanti.github.io/fp-ts/modules/Console.ts.html) module, with which `log("none")` represents an action but doesn't execute it.

**Reply by vinicius98s** [reply]
Yeah I see now, for some reason I was thinking that the `console.log` wouldn't be evaluated, thanks for the clarification
